memory-management/src/dynamic_layout.rs

```diff
@@ -0,0 +1,195 @@
+//! Dynamic memory layout containers
+
+/// An array of equally spaced values
```

Comment:
I think we should expand on the docs here, and ideally provide an example. `[T; N]` is equally spaced values too